src/compiler/checker.ts

```diff
@@ -2180,54 +2180,105 @@ namespace ts {
                         writer.writeKeyword(!(globalFlags & TypeFormatFlags.WriteOwnNameForAnyLike) && isTypeAny(type)
                             ? "any"
                             : (<IntrinsicType>type).intrinsicName);
+                        return;

```

Comment:
can you keep the `else if` for this change. it makes it easier to understand the extent of the change here.